qutip/core/data/convert.pyx

```diff
@@ -259,10 +261,31 @@ cdef class _to:
                     _converter(convert[::-1], to_t, from_t)
         for dispatcher in self.dispatchers:
             dispatcher.rebuild_lookup()
+        for dtype in self.dtypes:
+            self._str2type[dtype.__name__.lower()] = dtype
+
+    def str2type(self, arg):
+        if isinstance(arg, type):
+            return arg
+        if isinstance(arg, str):
+            if not arg.lower() in self._str2type:
```

Comment:
`x not in y` is supposedly better/faster than `not x in y` according to the Python docs, but I don't really know any detail on that